### PR TITLE
rimage: Add function to read section content with allocation

### PR DESCRIPTION
**Pull request by ktrzcinx**
In many functions there is need to read user section content
and check firmware version or write section content to some
manifest or dictionary. Previously to do it, in each function
was loop to search for proper section in proper module,
then allocate buffer, read content and check for possible error
between each step what is quite overwhelming. After change there
will be one function responsible for this task.

Signed-off-by: Karol Trzcinski <karolx.trzcinski@linux.intel.com>

### CODE REVIEW

File: rimage/elf.c [hunk 1]
```diff
@@ -89,18 +89,18 @@ static int elf_read_sections(struct image *image, struct module *module,
 		module->fw_ready_index = -EINVAL;
 	} else {
 		/* find manifest module data */
-		module->bss_index = elf_find_section(image, module, ".bss");
```
Comment (paulstelian97) [inline rimage/elf.c]: Typo in commit message: "Unused arguments _shouldn't_ be passed to function".

Comment (ktrzcinx) [reply rimage/elf.c]: changed

**Comment by mmaka1** [comment]
@lgirdwood @dcpleung Should this one wait for #2086 and target the new source tree?

**Comment by lgirdwood** [comment]
@mmaka1 yes, #2086 just pending a CI confirmation check from @zrombel .

**Comment by ktrzcinx** [comment]
Done in https://github.com/thesofproject/sof/pull/2649 and https://github.com/thesofproject/rimage/pull/11

**Pull request closed without merging**